src/task/CompoundTask.ts

```diff
@@ -40,11 +41,11 @@ export abstract class CompoundTask extends DFTask {
      * @param child
      *  A sub-task of this task.
      */
-    public handleCompletion(child: TaskBase): void {
+    public handleCompletion(child: TaskBase, executor?: TaskOrchestrationExecutor): void {
```

Comment:
Why is executor an optional argument?

Reply:
Because not all `CompoundTask`s would expect or utilize this argument. I could make the same change for all `CompoundTask`s, but I wanted to localize this change to be as small as possible. It is not optional for the `RetryTask` though, because an error will be thrown if it is not defined.

Reply:
Oh I see, I remember the context now. Makes sense. Could ou please add a comment noting this down? Maybe you can add it to the method-signature-level comment?

Reply:
Done